### DISCUSSION TITLE
Nginx proxy and session cookie

### DISCUSSION THREAD

**Discussion by Astagor** [edumeet/edumeet#827] [Install/Run/Config]
Hi all. I have a problem with Nginx as proxy and the session cookie.

I have set.
1. Nginx conf:
`location / {
                proxy_pass http://127.0.0.1:7780;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Fowarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Fowarded-Proto $scheme;
                proxy_set_header X-NginX-Proxy true;
                proxy_http_version 1.1;
                proxy_set_header Host $http_host;
                proxy_cache_bypass $http_upgrade;
        }`
2. server/config.js
`	httpOnly              : true,
        trustProxy            : 'loopback',
        listeningHost       :'localhost',`

However, in the app the session cookie is not set, as express does not respect that the first hop is a proxy to trust. According to [https://expressjs.com/en/guide/behind-proxies.html](url):
`The req.ip and req.ips values are populated with the list of addresses from X-Forwarded-For.`

In the logs I have following request headers:
` multiparty-meeting-server:WARN req.protocol= { 'x-real-ip': '79.238.37.46',
  'x-fowarded-for': '79.238.37.46',
  'x-fowarded-proto': 'https',
  'x-nginx-proxy': 'true',
  host: 'corona2.medvc.eu',
  connection: 'close',
  'cache-control': 'no-cache',
  'sec-fetch-site': 'none',
  'sec-fetch-mode': 'no-cors',
  'user-agent':
   'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.163 Safari/537.36',
  'accept-encoding': 'gzip, deflate, br',
  'accept-language': 'pl,en-US;q=0.9,en;q=0.8,de;q=0.7' } +1ms`

But when I print them in the server I get:

`  multiparty-meeting-server:WARN req.secure= false +0ms
  multiparty-meeting-server:WARN req.ip= 127.0.0.1 +0ms
  multiparty-meeting-server:WARN req.ips = [] +0ms`

So express does not set the session cookie.

I even changed in the server code to:
`app.set('trust proxy', function (ip) {
  logger.warn('ASTAGOR IP in TRUST PROXY=', ip);
  if (ip === '127.0.0.1')
  {
	  logger.warn('TRUE');
	  return true; // trusted IPs
  }
  else
  {
	  logger.warn('FALSE');
	  return false;
  }
});`

And I get a TRUE in the logs.

No idea why it is not working. In the end I have changed the code of the server, but this is an ugly hack:

`const session = expressSession({
  secret            : config.cookieSecret,
  name              : config.cookieName,
  resave            : true,
  saveUninitialized : true,
  proxy       : config.trustProxy ? true : false,
  store             : new RedisStore({ client: redisClient }),
  cookie            : {
    secure   : config.trustProxy ? false : true,
    httpOnly : true,
    maxAge   : 60 * 60 * 1000 // Expire after 1 hour since last request from user
  }
});`

So I set secure to false, when config.trustProxy is defined.

Any ideas?

**Comment by chiragmewada** [comment]
Hi, 
I am having same issue in iOS Safari. Did you found the solution?
